File: Python/underautomation/universal_robots/ssh/tools/sftp/sftp_file_attributes.py
import typing
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "..", "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Ssh.Tools.Sftp import SftpFileAttributes as sftp_file_attributes

class SftpFileAttributes:
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = sftp_file_attributes()
		else:
			self._instance = _internal
	def set_permissions(self, mode: int) -> None:
		self._instance.SetPermissions(mode)
	def get_bytes(self) -> typing.List[int]:
		return self._instance.GetBytes()
	@property
	def last_access_time(self) -> typing.Any:
		return self._instance.LastAccessTime
	@last_access_time.setter
	def last_access_time(self, value: typing.Any):
		self._instance.LastAccessTime = value
	@property
	def last_write_time(self) -> typing.Any:
		return self._instance.LastWriteTime
	@last_write_time.setter
	def last_write_time(self, value: typing.Any):
		self._instance.LastWriteTime = value
	@property
	def last_access_time_utc(self) -> typing.Any:
		return self._instance.LastAccessTimeUtc
	@last_access_time_utc.setter
	def last_access_time_utc(self, value: typing.Any):
		self._instance.LastAccessTimeUtc = value
	@property
	def last_write_time_utc(self) -> typing.Any:
		return self._instance.LastWriteTimeUtc
	@last_write_time_utc.setter
	def last_write_time_utc(self, value: typing.Any):
		self._instance.LastWriteTimeUtc = value
	@property
	def size(self) -> int:
		return self._instance.Size
	@size.setter
	def size(self, value: int):
		self._instance.Size = value
	@property
	def user_id(self) -> int:
		return self._instance.UserId
	@user_id.setter
	def user_id(self, value: int):
		self._instance.UserId = value
	@property
	def group_id(self) -> int:
		return self._instance.GroupId
	@group_id.setter
	def group_id(self, value: int):
		self._instance.GroupId = value
	@property
	def is_socket(self) -> bool:
		return self._instance.IsSocket
	@property
	def is_symbolic_link(self) -> bool:
		return self._instance.IsSymbolicLink
	@property
	def is_regular_file(self) -> bool:
		return self._instance.IsRegularFile
	@property
	def is_block_device(self) -> bool:
		return self._instance.IsBlockDevice
	@property
	def is_directory(self) -> bool:
		return self._instance.IsDirectory
	@property
	def is_character_device(self) -> bool:
		return self._instance.IsCharacterDevice
	@property
	def is_named_pipe(self) -> bool:
		return self._instance.IsNamedPipe
	@property
	def owner_can_read(self) -> bool:
		return self._instance.OwnerCanRead
	@owner_can_read.setter
	def owner_can_read(self, value: bool):
		self._instance.OwnerCanRead = value
	@property
	def owner_can_write(self) -> bool:
		return self._instance.OwnerCanWrite
	@owner_can_write.setter
	def owner_can_write(self, value: bool):
		self._instance.OwnerCanWrite = value
	@property
	def owner_can_execute(self) -> bool:
		return self._instance.OwnerCanExecute
	@owner_can_execute.setter
	def owner_can_execute(self, value: bool):
		self._instance.OwnerCanExecute = value
	@property
	def group_can_read(self) -> bool:
		return self._instance.GroupCanRead
	@group_can_read.setter
	def group_can_read(self, value: bool):
		self._instance.GroupCanRead = value
	@property
	def group_can_write(self) -> bool:
		return self._instance.GroupCanWrite
	@group_can_write.setter
	def group_can_write(self, value: bool):
		self._instance.GroupCanWrite = value
	@property
	def group_can_execute(self) -> bool:
		return self._instance.GroupCanExecute
	@group_can_execute.setter
	def group_can_execute(self, value: bool):
		self._instance.GroupCanExecute = value
	@property
	def others_can_read(self) -> bool:
		return self._instance.OthersCanRead
	@others_can_read.setter
	def others_can_read(self, value: bool):
		self._instance.OthersCanRead = value
	@property
	def others_can_write(self) -> bool:
		return self._instance.OthersCanWrite
	@others_can_write.setter
	def others_can_write(self, value: bool):
		self._instance.OthersCanWrite = value
	@property
	def others_can_execute(self) -> bool:
		return self._instance.OthersCanExecute
	@others_can_execute.setter
	def others_can_execute(self, value: bool):
		self._instance.OthersCanExecute = value
	@property
	def extensions(self) -> typing.Any:
		return self._instance.Extensions

File: Ssh/Tools/Sftp/SftpFileAttributes.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using System.Collections.Generic;
using System;

namespace Ssh.Tools.Sftp {
	/// <summary>
	/// Contains SFTP file attributes.
	/// </summary>
	public class SftpFileAttributes {

		/// <summary>
		/// Gets or sets the local time the current file or directory was last accessed.
		/// </summary>
		/// <returns>The local time that the current file or directory was last accessed.</returns>
		public DateTime LastAccessTime { get; set; }

		/// <summary>
		/// Gets or sets the local time when the current file or directory was last written to.
		/// </summary>
		/// <returns>The local time the current file was last written.</returns>
		public DateTime LastWriteTime { get; set; }

		/// <summary>
		/// Gets or sets the UTC time the current file or directory was last accessed.
		/// </summary>
		/// <returns>The UTC time that the current file or directory was last accessed.</returns>
		public DateTime LastAccessTimeUtc { get; set; }

		/// <summary>
		/// Gets or sets the UTC time when the current file or directory was last written to.
		/// </summary>
		/// <returns>The UTC time the current file was last written.</returns>
		public DateTime LastWriteTimeUtc { get; set; }

		/// <summary>
		/// Gets or sets the size, in bytes, of the current file.
		/// </summary>
		/// <returns>The size of the current file in bytes.</returns>
		public long Size { get; set; }

		/// <summary>
		/// Gets or sets file user id.
		/// </summary>
		/// <returns>File user id.</returns>
		public int UserId { get; set; }

		/// <summary>
		/// Gets or sets file group id.
		/// </summary>
		/// <returns>File group id.</returns>
		public int GroupId { get; set; }

		/// <summary>
		/// Gets a value indicating whether file represents a socket.
		/// </summary>
		/// <returns><code>true</code> if file represents a socket; otherwise, <code>false</code>.</returns>
		public bool IsSocket { get; }

		/// <summary>
		/// Gets a value indicating whether file represents a symbolic link.
		/// </summary>
		/// <returns><code>true</code> if file represents a symbolic link; otherwise, <code>false</code>.</returns>
		public bool IsSymbolicLink { get; }

		/// <summary>
		/// Gets a value indicating whether file represents a regular file.
		/// </summary>
		/// <returns><code>true</code> if file represents a regular file; otherwise, <code>false</code>.</returns>
		public bool IsRegularFile { get; }

		/// <summary>
		/// Gets a value indicating whether file represents a block device.
		/// </summary>
		/// <returns><code>true</code> if file represents a block device; otherwise, <code>false</code>.</returns>
		public bool IsBlockDevice { get; }

		/// <summary>
		/// Gets a value indicating whether file represents a directory.
		/// </summary>
		/// <returns><code>true</code> if file represents a directory; otherwise, <code>false</code>.</returns>
		public bool IsDirectory { get; }

		/// <summary>
		/// Gets a value indicating whether file represents a character device.
		/// </summary>
		/// <returns><code>true</code> if file represents a character device; otherwise, <code>false</code>.</returns>
		public bool IsCharacterDevice { get; }

		/// <summary>
		/// Gets a value indicating whether file represents a named pipe.
		/// </summary>
		/// <returns><code>true</code> if file represents a named pipe; otherwise, <code>false</code>.</returns>
		public bool IsNamedPipe { get; }

		/// <summary>
		/// Gets a value indicating whether the owner can read from this file.
		/// </summary>
		/// <returns><code>true</code> if owner can read from this file; otherwise, <code>false</code>.</returns>
		public bool OwnerCanRead { get; set; }

		/// <summary>
		/// Gets a value indicating whether the owner can write into this file.
		/// </summary>
		/// <returns><code>true</code> if owner can write into this file; otherwise, <code>false</code>.</returns>
		public bool OwnerCanWrite { get; set; }

		/// <summary>
		/// Gets a value indicating whether the owner can execute this file.
		/// </summary>
		/// <returns><code>true</code> if owner can execute this file; otherwise, <code>false</code>.</returns>
		public bool OwnerCanExecute { get; set; }

		/// <summary>
		/// Gets a value indicating whether the group members can read from this file.
		/// </summary>
		/// <returns><code>true</code> if group members can read from this file; otherwise, <code>false</code>.</returns>
		public bool GroupCanRead { get; set; }

		/// <summary>
		/// Gets a value indicating whether the group members can write into this file.
		/// </summary>
		/// <returns><code>true</code> if group members can write into this file; otherwise, <code>false</code>.</returns>
		public bool GroupCanWrite { get; set; }

		/// <summary>
		/// Gets a value indicating whether the group members can execute this file.
		/// </summary>
		/// <returns><code>true</code> if group members can execute this file; otherwise, <code>false</code>.</returns>
		public bool GroupCanExecute { get; set; }

		/// <summary>
		/// Gets a value indicating whether the others can read from this file.
		/// </summary>
		/// <returns><code>true</code> if others can read from this file; otherwise, <code>false</code>.</returns>
		public bool OthersCanRead { get; set; }

		/// <summary>
		/// Gets a value indicating whether the others can write into this file.
		/// </summary>
		/// <returns><code>true</code> if others can write into this file; otherwise, <code>false</code>.</returns>
		public bool OthersCanWrite { get; set; }

		/// <summary>
		/// Gets a value indicating whether the others can execute this file.
		/// </summary>
		/// <returns><code>true</code> if others can execute this file; otherwise, <code>false</code>.</returns>
		public bool OthersCanExecute { get; set; }

		/// <summary>
		/// Gets or sets the extensions.
		/// </summary>
		/// <returns>The extensions.</returns>
		public IDictionary<string, string> Extensions { get; }

		/// <summary>
		/// Sets the permissions.
		/// </summary>
		/// <param name="mode">The mode.</param>
		public void SetPermissions(short mode)
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Returns a byte array representing the current <xref href="UnderAutomation.UniversalRobots.Ssh.Tools.Sftp.SftpFileAttributes" data-throw-if-not-resolved="false"></xref>.
		/// </summary>
		/// <returns>A byte array representing the current <xref href="UnderAutomation.UniversalRobots.Ssh.Tools.Sftp.SftpFileAttributes" data-throw-if-not-resolved="false"></xref>.</returns>
		public byte[] GetBytes()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}
	}
}
